### ISSUE TITLE
Beacon crashes when trying to attach a video

### ISSUE THREAD

**Issue by bvirlet** [helpscout/beacon-ios-sdk#63]
* Beacon install method: [x] Carthage, [ ] CocoaPods, [ ] Manual

**Actual outcome**

- Select **Ask**
- Select **Take photo**
- Switch to **Video**
- App crashes

**Expected outcome**
- App should not crash

Thanks!

**Comment by shaunmcilroy** [comment]
Hi @bvirlet can you please confirm for us which version of the SDK you are currently using, and where possible can you record the above steps and the crash in action?

Thank you.

**Comment by bvirlet** [comment]
`github "helpscout/beacon-ios-sdk" "1.0.5"`

This is very minor for us, it can be reproduced with above steps.

**Comment by shaunmcilroy** [comment]
Thanks for the note, I've been unable to reproduce this using v1.0.5 of our SDK embedded in an app on an iPhone 11 ([video](https://c.hlp.sc/KouBzADn)). The steps I'm following are:
- Click Get in Touch
- Fill out form details
- Click attachment icon
- Take Photo
- Switch to Video
- Record and attach
- Submit

When you say you select "Ask" do you refer to the steps shown in the attached video? We have the Ask screen only on V2 of Beacon ([web example](https://c.hlp.sc/qGuKA7NJ)), which 1.0.5 doesn't support at this time.

I would recommend updating to the latest version of the SDK as we've released a number of bug fixes and changes since v1.0.5 released. I appreciate that you'd previously mentioned that you'd prefer not to use the Chat features (or dependencies) included with v2.0.0+ but we have no current plans to offer a version of Beacon that doesn't include these features at this time. 

Should that change in the future we'll let you know.

**Comment by shaunmcilroy** [comment]
Hi @bvirlet I'm following up to see if you had any other information to share as per my previous note? Let us know if there's anything else you're able to confirm or rule out and we'll decide on next steps for this issue. Thanks!

**Comment by bvirlet** [comment]
Hi Shaun,

The issue can be reproduced with our app Genius Scan. I digged deeper and I get the following error message in our logs:
```
Genius Scan[39019:5871928] [access] This app has crashed because it attempted to access privacy-sensitive data without a usage description.  The app's Info.plist must contain an NSMicrophoneUsageDescription key with a string value explaining to the user how the app uses this data.
```

So probably just need to add a note to add that permission in the documentation: https://developer.helpscout.com/beacon-2/ios/. But that doesn't explain why you wouldn't reproduce this issue?

**Comment by shaunmcilroy** [comment]
Hi @bvirlet 

If you've already added the string to your info.plist then it should not crash from my testing. The test environment I used was our own Help Scout app which currently runs the same version (v1.0.5) of the SDK and includes [those required strings](https://c.hlp.sc/qGuKdyrn) noted here on [the Apple dev site](https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CocoaKeys.html#//apple_ref/doc/uid/TP40009251-SW74) which is likely why it didn't crash on my side. When I downloaded and ran both the Help Scout and Genius Scan apps one after each other I was able to reproduce your issue though only in your app as seen in this video: https://c.hlp.sc/P8ueEgR9

Have you added the required NSMicrophoneUsageDescription key to your info.plist and is the Take Photo video recorder working correctly now?

**Comment by bvirlet** [comment]
Hi @shaunmcilroy,

Thanks - yes, adding this string fixes the issue. I thought it was strange you couldn't reproduce the issue because I looked at the Beacon Example app and it doesn't include this string.

This was really not important for us, but I reported for the sake of completeness as our QA testers found it. As I mentioned, maybe the Beacon iOS documentation should mention this string alongside NSCameraUsageDescription it already mentions. Otherwise, consider this closed for me.
